server/mqtt.go

```diff
@@ -4309,14 +4310,13 @@ func (c *client) mqttHandlePubRetain() {
 		// MQTT Server MUST make DBIRTH messages available on a topic of the
 		// form:
 		// $sparkplug/certificates/namespace/group_id/DBIRTH/edge_node_id/device_id
-		rm.Topic = fmt.Sprintf(sparkbCertificates+"/%s/%s/%s", groupID, typ, nodeID)
-		if deviceID != _EMPTY_ {
-			rm.Topic += "/" + deviceID
-		}
-		byteSubject, _ := mqttTopicToNATSPubSubject([]byte(rm.Topic))
+		topic := append(sparkbCertificatesTopicPrefix, pp.topic...)
+		subject, _ := mqttTopicToNATSPubSubject(topic)
+		rm.Topic = string(topic)
+		rm.Subject = string(subject)
 
 		// will use to save the retained message.
-		key = string(byteSubject)
+		key = string(subject)
 
 		// Store the retained message with the RETAIN flag set.
 		rm.Flags |= mqttPubFlagRetain
```

Comment:
This seem to indicate that they may not have the flag on inbound, so again, don't we need to make sure that they don't have it?

Reply:
Resolving this thread in favor of ^^